src/dc_receive_imf.rs

```diff
@@ -1378,6 +1365,27 @@ async fn create_or_lookup_group(
     Ok((chat_id, chat_id_blocked))
 }
 
+fn try_getting_grpid(mime_parser: &MimeMessage) -> String {
+    if let Some(optional_field) = mime_parser.get(HeaderDef::ChatGroupId) {
+        return optional_field.clone();
+    }
+
+    if let Some(extracted_grpid) = mime_parser
+        .get(HeaderDef::MessageId)
+        .and_then(|value| dc_extract_grpid_from_rfc724_mid(&value))
+    {
+        return extracted_grpid.to_string();
```

Comment:
Why are we looking for the group id in the `MessageId`? As I understood it, Delta-emails will contain a `ChatGroupId` and non-delta-emails won't contain a group id in the `MessageId`

Reply:
I also thought about it when changed this code, see issue #1568. Message-ID lookup can be probably dropped already.

Reply:
BTW we should keep in mind that we still need group ids stored in MessageId's for prefetching (so that we get to know that a blocked contact removed us from a group)